src/io/utils.rs

```diff
@@ -200,34 +204,107 @@ where
 	Ok(res)
 }
 
+/// Read `OutputSweeper` state from the store.
+pub(crate) fn read_output_sweeper<K: KVStore + Send + Sync>(
+	broadcaster: Arc<Broadcaster>, fee_estimator: Arc<FeeEstimator>,
+	chain_data_source: Arc<ChainSource>, keys_manager: Arc<KeysManager>, kv_store: Arc<K>,
+	logger: Arc<FilesystemLogger>,
+) -> Result<Sweeper<K>, std::io::Error> {
+	let mut reader = Cursor::new(kv_store.read(
+		OUTPUT_SWEEPER_PERSISTENCE_PRIMARY_NAMESPACE,
+		OUTPUT_SWEEPER_PERSISTENCE_SECONDARY_NAMESPACE,
+		OUTPUT_SWEEPER_PERSISTENCE_KEY,
+	)?);
+	let args = (
+		broadcaster,
+		fee_estimator,
+		Some(chain_data_source),
+		Arc::clone(&keys_manager),
+		keys_manager,
+		kv_store,
+		logger.clone(),
+	);
+	OutputSweeper::read(&mut reader, args).map_err(|e| {
+		log_error!(logger, "Failed to deserialize OutputSweeper: {}", e);
+		std::io::Error::new(std::io::ErrorKind::InvalidData, "Failed to deserialize OutputSweeper")
+	})
+}
+
 /// Read previously persisted spendable output information from the store.
-pub(crate) fn read_spendable_outputs<K: KVStore + Sync + Send, L: Deref>(
-	kv_store: Arc<K>, logger: L,
-) -> Result<Vec<SpendableOutputInfo>, std::io::Error>
+pub(crate) fn migrate_deprecated_spendable_outputs<K: KVStore + Sync + Send, L: Deref>(
+	sweeper: Arc<Sweeper<K>>, kv_store: Arc<K>, logger: L,
+) -> Result<(), std::io::Error>
 where
 	L::Target: Logger,
 {
-	let mut res = Vec::new();
+	let best_block = sweeper.current_best_block();
 
 	for stored_key in kv_store.list(
-		SPENDABLE_OUTPUT_INFO_PERSISTENCE_PRIMARY_NAMESPACE,
-		SPENDABLE_OUTPUT_INFO_PERSISTENCE_SECONDARY_NAMESPACE,
+		DEPRECATED_SPENDABLE_OUTPUT_INFO_PERSISTENCE_PRIMARY_NAMESPACE,
+		DEPRECATED_SPENDABLE_OUTPUT_INFO_PERSISTENCE_SECONDARY_NAMESPACE,
 	)? {
 		let mut reader = Cursor::new(kv_store.read(
-			SPENDABLE_OUTPUT_INFO_PERSISTENCE_PRIMARY_NAMESPACE,
-			SPENDABLE_OUTPUT_INFO_PERSISTENCE_SECONDARY_NAMESPACE,
+			DEPRECATED_SPENDABLE_OUTPUT_INFO_PERSISTENCE_PRIMARY_NAMESPACE,
+			DEPRECATED_SPENDABLE_OUTPUT_INFO_PERSISTENCE_SECONDARY_NAMESPACE,
 			&stored_key,
 		)?);
-		let output = SpendableOutputInfo::read(&mut reader).map_err(|e| {
+		let output = DeprecatedSpendableOutputInfo::read(&mut reader).map_err(|e| {
 			log_error!(logger, "Failed to deserialize SpendableOutputInfo: {}", e);
 			std::io::Error::new(
 				std::io::ErrorKind::InvalidData,
 				"Failed to deserialize SpendableOutputInfo",
 			)
 		})?;
-		res.push(output);
+		let descriptors = vec![output.descriptor.clone()];
+		let spend_delay = Some(best_block.height + 2);
+		sweeper.track_spendable_outputs(descriptors, output.channel_id, true, spend_delay);
+		if let Some(tracked_spendable_output) =
+			sweeper.tracked_spendable_outputs().iter().find(|o| o.descriptor == output.descriptor)
+		{
+			match tracked_spendable_output.status {
+				OutputSpendStatus::PendingInitialBroadcast { delayed_until_height } => {
```

Comment:
It isn't clear to me why this is the only state we would expect. Could you add a comment somewhere describing the migration?